pkg/inventory/db.go

```diff
@@ -200,25 +170,119 @@ func (db *DB) GetResources(ctx context.Context) <-chan []resourceapi.Device {
-	// expose the real interface name as an attribute in case it is normalized.
-	device.Attributes["dra.net/ifName"] = resourceapi.DeviceAttribute{StringValue: &ifName}
-
-	linkType := link.Type()
-	linkAttrs := link.Attrs()
-
-	// identify the namespace holding the link as the other end of a veth pair
-	netnsid := link.Attrs().NetNsID
-	if podName := db.GetPodName(netnsid); podName != "" {
-		device.Attributes["dra.net/pod"] = resourceapi.DeviceAttribute{StringValue: &podName}
+
+	for _, pciDev := range pci.Devices {
+		if !isNetworkDevice(pciDev) {
+			continue
+		}
+		device := resourceapi.Device{
+			Name:       names.NormalizePCIAddress(pciDev.Address),
+			Attributes: make(map[resourceapi.QualifiedName]resourceapi.DeviceAttribute),
+			Capacity:   make(map[resourceapi.QualifiedName]resourceapi.DeviceCapacity),
+		}
+		device.Attributes[apis.AttrPCIAddress] = resourceapi.DeviceAttribute{StringValue: &pciDev.Address}
+		if pciDev.Vendor != nil {
+			device.Attributes[apis.AttrPCIVendor] = resourceapi.DeviceAttribute{StringValue: &pciDev.Vendor.Name}
+		}
+		if pciDev.Product != nil {
+			device.Attributes[apis.AttrPCIDevice] = resourceapi.DeviceAttribute{StringValue: &pciDev.Product.Name}
+		}
+		if pciDev.Subsystem != nil {
+			device.Attributes[apis.AttrPCISubsystem] = resourceapi.DeviceAttribute{StringValue: &pciDev.Subsystem.ID}
+		}
+
+		if pciDev.Node != nil {
+			device.Attributes[apis.AttrNUMANode] = resourceapi.DeviceAttribute{IntValue: ptr.To(int64(pciDev.Node.ID))}
+		}
+
+		pcieRootAttr, err := deviceattribute.GetPCIeRootAttributeByPCIBusID(pciDev.Address)
+		if err != nil {
+			klog.Infof("Could not get pci root attribute: %v", err)
+		} else {
+			device.Attributes[pcieRootAttr.Name] = pcieRootAttr.Value
+		}
+		devices = append(devices, device)
 	}
+	return devices
+}
+
+func (db *DB) discoverNetworkInterfaces(pciDevices []resourceapi.Device) []resourceapi.Device {
+	links, err := netlink.LinkList()
+	if err != nil {
+		klog.Errorf("Could not list network interfaces: %v", err)
+		return pciDevices
+	}
+
+	pciDeviceMap := make(map[string]*resourceapi.Device)
+	for i := range pciDevices {
+		pciDeviceMap[pciDevices[i].Name] = &pciDevices[i]
+	}
+
+	otherDevices := []resourceapi.Device{}
+
+	for _, link := range links {
+		ifName := link.Attrs().Name
+		if ignoredInterfaceNames.Has(ifName) {
+			klog.V(4).Infof("Network Interface %s is in the list of ignored interfaces, excluding it from discovery", ifName)
+			continue
+		}
+
+		// skip loopback interfaces
+		if link.Attrs().Flags&net.FlagLoopback != 0 {
+			klog.V(4).Infof("Network Interface %s is a loopback interface, excluding it from discovery", ifName)
+			continue
+		}
+
+		pciAddr := link.Attrs().ParentDev
+
+		var device *resourceapi.Device
+
+		if pciAddr != "" {
+			// It's a PCI device.
+			normalizedAddress := names.NormalizePCIAddress(pciAddr)
+			var exists bool
+			device, exists = pciDeviceMap[normalizedAddress]
+			if !exists {
```

Comment:
Should we even be adding here? This feels like a bug since we should be adding it elsewhere.